gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -1309,4 +1319,94 @@ var gpii = fluid.registerNamespace("gpii");
+     *    session will be retrieved
+     * @param {String} [forcePrefsSetName] - [optional] A prefsSet name to be unconditionally selected
+     * @return {Promise} If the preferences set change causes an actual change in the system (i.e. change of
+     *    activePrefsSet and changes to the applied configuration) a promise of an update to the system
+     *    configuration is returned. If the call to prefsSetChanged does not result in a different
+     *    active preferences set or no user is logged in, undefined is returned.
+     **/
+    gpii.lifecycleManager.prefsSetNameChanged = function (that, forcePrefsSetName) {
+        // find logged in users
+        var activeSession = that.getSession();
+        var activeGpiiKey = activeSession.model.gpiiKey;
+
+        // if no one is logged in, or, a system reserved key is logged in, do nothing
+        if (!activeGpiiKey || gpii.isReservedGpiiKey(activeGpiiKey)) {
+            return;
+        }
+        var fullPayload = fluid.extend(true, {}, activeSession.model);
+        var oldPrefsSetName = fullPayload.activePrefsSetName;
+
+        // find and update preferences set name:
+        that.updateActivePrefsSetName(fullPayload, forcePrefsSetName);
+
+        if (oldPrefsSetName === fullPayload.activePrefsSetName) {
+            fluid.log(fluid.logLevel.TRACE, "lifecycleManager: Same preferences set as before (" + oldPrefsSetName + ") so doing nothing");
+            return;
+        }
+        fluid.log("lifecycleManager: Context changed from old context \"", oldPrefsSetName, "\" to new active context \"" + fullPayload.activePrefsSetName, "\" for the GPII key: ", activeGpiiKey);
+        // Update the context name immediately so we don't produce a model update from the lifecycleManager with a
+        // partially applied set of solutions from the new context, confusing, e.g., the PSPChannel
+        // This is required because of https://issues.fluidproject.org/browse/FLUID-6208
+        activeSession.applier.change("activePrefsSetName", fullPayload.activePrefsSetName);
+
+        // Delete the old appliedSolutions as the new structure will be updated directly onto activeSession.model
+        // by lifecycleManager.update() called 2 lines later
+        delete fullPayload.appliedSolutions;
+
+        that.addLifecycleInstructionsToPayload(fullPayload);
+        var response = that.update(fullPayload);
+
+        response.then(function () {
+            // Delete these this to avoid the 'lifecycleInstructions' block being attached
+            // to the sessionState in the next line. LifecycleInstructions is only used in the
+            // "finalPayload" as instructions on how to configure the system. once configured,
+            // the applied settings are stored in 'activeConfiguration'
+            delete fullPayload.lifecycleInstructions;
+            // TODO: This is pretty rubbish, why isn't the natural action of lifecycleManager.update good enough here?
```

Comment:
This comment should be removed. start() and update() only apply settings to solutions but don't touch session model.

Reply:
So, the commenter's question has been answered.  I wonder if they know...

Sure, I'll remove it.